packages/billing/stripe-webhook-handlers/subscription-deleted.ts

```diff
@@ -71,9 +73,43 @@ export async function handleSubscriptionDeleted(
 
     // This happens when the payment method fails for a couple of times and then Stripe subscription gets expired.
     if (newStatus === 'expired') {
-      // TODO: Put the user back on the free plan (by calling Stripe API). Will be handled in CS-7466
+      await subcribeUserToFreePlan(dbAdapter, event.data.object.customer);
     }
 
     await markStripeEventAsProcessed(dbAdapter, event.id);
   });
 }
+
+async function subcribeUserToFreePlan(
+  dbAdapter: DBAdapter,
+  stripeCustomerId: string,
+) {
+  try {
+    let stripe = getStripe();
+    let freePlan = await getPlanByMonthlyPrice(dbAdapter, 0);
+    if (!freePlan) {
+      console.error('Free plan is not found');
+      return;
+    }
+    let prices = await stripe.prices.list({
+      product: freePlan.stripePlanId,
+      active: true,
+    });
+    if (!prices.data[0]) {
+      console.error('No price found for free plan');
+      return;
+    }
+
+    await stripe.subscriptions.create({
```

Comment:
Does this charge the user (with 0 amount) immediately? What happens if the user's payment method is invalid? What are the possible errors here, returned by Stripe?

Reply:
Did you confirm the subsequent invoice.payment_succeeded webhook activated the free plan?

Reply:
I mean in the boxel app - did the webhook create a new `subscription`, `subscription_cycle`,... with the new free plan?

Reply:
I just tested it, and it worked. One question: don’t we need to ensure the user is always subscribed to a plan? If so, why don’t we subscribe the user to a free plan when their subscription is canceled, not just when it expires?

Reply:
> I just tested it, and it worked

👍 

> why don’t we subscribe the user to a free plan when their subscription is canceled

Because even on the stripe free plan, users will keep receiving the monthly $0 invoice. So if they cancel we must respect their decision to not be subscribed anymore.